server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServices.java

```diff
@@ -444,10 +441,10 @@ private CompositeToken createCompositeToken(String tokenId,
         compositeToken.setValue(token);
         BaseClientDetails clientDetails = (BaseClientDetails) clientDetailsService.loadClientByClientId(clientId);
 
-        if (idTokenGranter.shouldSendIdToken(userId, clientDetails, requestedScopes, grantType)) {
+        if (idTokenGranter.shouldSendIdToken(user == null ? null : user.getId(), clientDetails, requestedScopes, grantType)) {
```

Comment:
I don't think this null check is needed.

`createCompositeToken` has two call sites `UaaTokenServices#refreshAccessToken` and `UaaTokenServices#createAccessToken` both of which perform a lookup by id for the user and pass the result to this method.

That means that the user object passed into this method cannot be null, correct?

Reply:
refactored code